presto-druid/src/test/java/com/facebook/presto/druid/TestDruidQueryBase.java

```diff
@@ -0,0 +1,219 @@
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.spi.type.DoubleType.DOUBLE;
+import static com.facebook.presto.spi.type.VarcharType.VARCHAR;
+import static com.facebook.presto.sql.analyzer.ExpressionAnalyzer.getExpressionTypes;
+import static com.facebook.presto.testing.TestingConnectorSession.SESSION;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.util.Locale.ENGLISH;
+import static java.util.Objects.requireNonNull;
+import static java.util.function.Function.identity;
+import static java.util.stream.Collectors.toMap;
+
+public class TestDruidQueryBase
+{
+    protected static final TypeManager typeManager = new TypeRegistry();
+    protected static final FunctionManager functionMetadataManager = new FunctionManager(typeManager, new BlockEncodingManager(typeManager), new FeaturesConfig());
+    protected static final StandardFunctionResolution standardFunctionResolution = new FunctionResolution(functionMetadataManager);
+
+    protected static ConnectorId druidConnectorId = new ConnectorId("id");
+    protected static DruidTableHandle realtimeOnlyTable = new DruidTableHandle("schema", "realtimeOnly", Optional.empty());
+    protected static DruidTableHandle hybridTable = new DruidTableHandle("schema", "hybrid", Optional.empty());
+    protected static DruidColumnHandle regionId = new DruidColumnHandle("regionId", BIGINT, REGULAR);
+    protected static DruidColumnHandle city = new DruidColumnHandle("city", VARCHAR, REGULAR);
+    protected static final DruidColumnHandle fare = new DruidColumnHandle("fare", DOUBLE, REGULAR);
+    protected static final DruidColumnHandle secondsSinceEpoch = new DruidColumnHandle("secondsSinceEpoch", BIGINT, REGULAR);
+
+    protected static final Metadata metadata = MetadataManager.createTestMetadataManager();
+
+    protected final DruidConfig druidConfig = new DruidConfig();
+
+    protected static final Map<VariableReferenceExpression, DruidQueryGeneratorContext.Selection> testInput = ImmutableMap.of(
+            new VariableReferenceExpression("regionid", BIGINT), new DruidQueryGeneratorContext.Selection("regionId", TABLE_COLUMN),
+            new VariableReferenceExpression("city", VARCHAR), new DruidQueryGeneratorContext.Selection("city", TABLE_COLUMN),
+            new VariableReferenceExpression("fare", DOUBLE), new DruidQueryGeneratorContext.Selection("fare", TABLE_COLUMN),
+            new VariableReferenceExpression("totalfare", DOUBLE), new DruidQueryGeneratorContext.Selection("(fare + trip)", DERIVED),
+            new VariableReferenceExpression("secondssinceepoch", BIGINT), new DruidQueryGeneratorContext.Selection("secondsSinceEpoch", TABLE_COLUMN));
+
+    protected final TypeProvider typeProvider = TypeProvider.fromVariables(testInput.keySet());
+
+    protected static class SessionHolder
+    {
+        private final ConnectorSession connectorSession;
+        private final Session session;
```

Comment:
Do we really need both sessions?

Reply:
yep, need connectorSession for timeZoneKey, and Session for ExpressionAnalyzer